### DISCUSSION TITLE
how to install latest cognifly from code?

### DISCUSSION THREAD

**Discussion by lida2003** [thecognifly/cognifly-python#4] [General]
Well, I'm tring to install cognifly on my RPI (Google AIY board), which pre-installed with [AIY Kits Release 2021-04-02](https://github.com/google/aiyprojects-raspbian/releases/tag/v20210402).

When I try to use latest code, I found it's quite difficult for me to install or run the program.

Since all docs are linked to install version 0.3.1 (released version). 

a) [readme-installation](https://github.com/thecognifly/cognifly-python/tree/yb/stable#installation) suggests using [version 0.3.1](https://pypi.org/project/cognifly/) from https://pypi.org/project/cognifly/
b) [git code - setup.py](https://github.com/thecognifly/cognifly-python/blob/yb/stable/setup.py) also suggests using [version 0.3.1](https://github.com/thecognifly/cognifly-python/archive/refs/tags/v0.3.1.tar.gz) from https://github.com/thecognifly/cognifly-python/archive/refs/tags/v0.3.1.tar.gz 

Is there any guide or links available for me to  run from the souce code? Which might have pip scripts to install a basic environment, including [YAMSPy](https://github.com/thecognifly/YAMSPy)?

BTW,  [YAMSPy](https://github.com/thecognifly/YAMSPy) which branch should I use `yb/stable`?

**Comment by yannbouteiller** [comment]
The issue is that you are trying to install the version that is currently under development, the documentation is not yet available. It should be in 2 weeks or so, though, as `yb/stable` will be pulled to `master`.

In the meantime, yes, you first need to install the `yb/stable` branch of Yamspy and INAV 6. Then you will need my current `.txt` configuration file, hold on.

**Reply by lida2003** [reply]
Is there any [setup.py](https://docs.python.org/3/distutils/setupscript.html) script that can use following commands:
```
# python3 setup.py build
# python3 setup.py install
```

Which build from git source code and install application to /usr/python/bin ?

PS: Current setup.py just download 0.3.1 released version and install to /usr/python/bin.

And i checked readme of yamspy, it suggest below commands. How to use my local git code (I don't have to brother which branch or git commit id I'm using.)

`$ sudo pip3 install git+https://github.com/ricardodeazambuja/yamspy --upgrade`

**Reply by yannbouteiller** [reply]
No, for both yamspy and cognifly-python, you need to checkout the `yb/stable` branch, pull the last version, and install locally by executing `pip install -e .` where the `setup.py` file is.

**Reply by lida2003** [reply]
OK. It seems to be installed and cognifly-controller (sudo required)

I really curious about where yamspy is installed?


```
pi@PiCam:~/YAMSPy $ sudo ls -l /usr/local/bin/
total 552
-rwxr-xr-x 1 root root    522 Jan 15 16:50 cognifly-controller
-rwxr-xr-x 1 root root    221 Jan 15 11:07 f2py
-rwxr-xr-x 1 root root    221 Jan 15 11:07 f2py3
-rwxr-xr-x 1 root root    221 Jan 15 11:07 f2py3.7
-rwxr-xr-x 1 root root   6058 Apr  3  2021 fixup_speech_v1_keywords.py
-rwxr-xr-x 1 root root   6058 Apr  3  2021 fixup_speech_v1p1beta1_keywords.py
-rwxr-xr-x 1 root root    236 Apr  3  2021 google-oauthlib-tool
-rwxr-xr-x 1 root root    214 Apr  3  2021 nosetests
-rwxr-xr-x 1 root root    214 Apr  3  2021 nosetests-3.4
-rwxr-xr-x 1 root root 490740 Nov 16  2020 protoc
drwxr-xr-x 2 root root   4096 Apr  3  2021 __pycache__
-rwxr-xr-x 1 root root    215 Apr  3  2021 pyrsa-decrypt
-rwxr-xr-x 1 root root    215 Apr  3  2021 pyrsa-encrypt
-rwxr-xr-x 1 root root    213 Apr  3  2021 pyrsa-keygen
-rwxr-xr-x 1 root root    236 Apr  3  2021 pyrsa-priv2pub
-rwxr-xr-x 1 root root    209 Apr  3  2021 pyrsa-sign
-rwxr-xr-x 1 root root    213 Apr  3  2021 pyrsa-verify

```

- YAMSPy  installation

```
pi@PiCam:~/YAMSPy $ pip install -e .
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple, http://mirrors.aliyun.com/pypi/simple/
Obtaining file:///home/pi/YAMSPy
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pyserial in /usr/lib/python3/dist-packages (from yamspy==0.3.4) (3.4)
Installing collected packages: yamspy
  Running setup.py develop for yamspy
Successfully installed yamspy
pi@PiCam:~/YAMSPy $ git log -n 1
commit 59be1eaad4a344c7fefb0be746fbc46c1f0d48e6 (HEAD -> yb/stable, origin/yb/stable)
Author: Yann Bouteiller <yann.bouteiller@hotmail.fr>
Date:   Mon Nov 28 16:03:02 2022 -0500

    flush defaults to False

```

- cognifly-controller installation

```
pi@PiCam:~/cognifly-python $ sudo pip install -e .
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Obtaining file:///home/pi/cognifly-python
Requirement already satisfied: Pillow in /usr/lib/python3/dist-packages (from cognifly==0.3.1) (5.4.1)
Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from cognifly==0.3.1) (1.21.6)
Requirement already satisfied: opencv-python in /usr/local/lib/python3.7/dist-packages (from cognifly==0.3.1) (4.7.0.68)
Requirement already satisfied: picamera in /usr/lib/python3/dist-packages (from cognifly==0.3.1) (1.13)
Requirement already satisfied: psutil in /usr/lib/python3/dist-packages (from cognifly==0.3.1) (5.5.1)
Requirement already satisfied: yamspy in /home/pi/YAMSPy (from cognifly==0.3.1) (0.3.4)
Requirement already satisfied: pyserial in /usr/lib/python3/dist-packages (from yamspy->cognifly==0.3.1) (3.4)
Installing collected packages: cognifly
  Found existing installation: cognifly 0.3.1
    Can't uninstall 'cognifly'. No files were found to uninstall.
  Running setup.py develop for cognifly
Successfully installed cognifly
pi@PiCam:~/cognifly-python $ git log -n 1
commit b2e29a14dd842dd4e50ec9b1d411970d90b7b403 (HEAD -> yb/stable, origin/yb/stable)
Author: Yann Bouteiller <yann.bouteiller@hotmail.fr>
Date:   Fri Dec 2 15:11:30 2022 -0500

    testing

```

**Comment by yannbouteiller** [comment]
[stable_inav6_config_custom_estimator.zip](https://github.com/thecognifly/cognifly-python/files/10417982/stable_inav6_config_custom_estimator.zip)

Attached you can find my current configuration file. However, this one works with a custom estimator that overrides all sensors. To make it work without a custom estimator, you will need to go to inav-configurator and adapt the sensor configuration to your setting.

**Reply by lida2003** [reply]
Thanks.

**Reply by lida2003** [reply]
OK, Now I have time to test it.

I have oddicity 2.5inch frame with inav 6.0 firmware.

I'm NOT sure how to configure. Currently, I have serial0 GPIO14/15 connected to FC and configured MSP port.
 [schematis of raspberry pi ZERO W](https://datasheets.raspberrypi.com/rpizero/raspberry-pi-zero-w-reduced-schematics.pdf) shows .
![图片](https://user-images.githubusercontent.com/1296171/212592823-ab5d608a-a9e2-4559-83f0-e407c7415e25.png)

BUT I'm NOT sure about configurations in Modes tab:
1. arm channel
2. posHold/altHold/surface
3. Flight modes

**Reply by lida2003** [reply]
I notice the reminder:

```
# DEFAULT_CH5 = 1000  # DISARMED (1000) / ARMED (1800)
# DEFAULT_CH6 = 1500  # ANGLE (1000) / NAV_POSHOLD (1500) <= 1300: NAV ALTHOLD
# DEFAULT_CH7 = 1000  # FAILSAFE (1800)
# DEFAULT_CH8 = 1000  # HEADING HOLD (1800)
```

Quite confused about CH6, and please help to check is my setting on inav configurator is correct: 

1) CH5,
![图片](https://user-images.githubusercontent.com/1296171/219225743-9fc817b6-87cd-4d97-9df6-8e81a298a2fc.png)

2) CH6, I'm NOT sure, how to configure???
![图片](https://user-images.githubusercontent.com/1296171/219225316-ef670f3d-1693-45e9-bce6-4497304f3956.png)

3) CH7, 
![图片](https://user-images.githubusercontent.com/1296171/219225656-a582b705-cdc7-4639-8629-f1afdcd4db29.png)

4)CH8
![图片](https://user-images.githubusercontent.com/1296171/219225484-32d02272-a851-47e5-bbe2-520f486860fa.png)

**Comment by lida2003** [comment]
Or as code suggested:

DEFAULT_AUX2 = NAV_POSHOLD_MODE  # Angle (1000) / NAV_ALTHOLD (1500) / NAV_POSHOLD (1800)
